planning/obstacle_avoidance_planner/include/obstacle_avoidance_planner/node.hpp

```diff
@@ -96,14 +97,20 @@ class ObstacleAvoidancePlanner : public rclcpp::Node
   rclcpp::Publisher<Trajectory>::SharedPtr debug_extended_traj_pub_;
   rclcpp::Publisher<MarkerArray>::SharedPtr debug_markers_pub_;
   rclcpp::Publisher<StringStamped>::SharedPtr debug_calculation_time_pub_;
+  rclcpp::Publisher<Trajectory>::SharedPtr debug_traj_pub_;
 
   // parameter callback
   rcl_interfaces::msg::SetParametersResult onParam(
     const std::vector<rclcpp::Parameter> & parameters);
   OnSetParametersCallbackHandle::SharedPtr set_param_res_;
 
-  // subscriber callback function
-  void onPath(const Path::SharedPtr);
+  // tf
+  std::shared_ptr<tf2_ros::Buffer> map_odom_tf_buffer_;
+  std::shared_ptr<tf2_ros::TransformListener> map_odom_tf_listener_;
```

Comment:
Reason for making this a `std::shared_ptr` (as opposed to a `std::unique_ptr` or not having a pointer at all)?

Reply:
It seems like all of these three work for tf2_ros::Buffer so I dont have a specific reason. I will take which ever you recommend. To my knowledge share_ptr and unique_ptr will automatically the object and unique_ptr cannot be copied or shared.

Reply:
The main difference between `std::shared_ptr` and `std::unique_ptr` is that `std::shared_ptr` maintains an internal reference counter which tracks how many instances of the object are being referenced. It's mostly for multi-threaded applications. And I realized that `tf2_ros::TransformListener` requires a `tf2_ros::Buffer` in its constructor so using a pointer wrapper makes sense. However, I recommend `std::unique_ptr` in this case.

Reply:
Make sense! I have changed it to unique_ptr for tf2_ros:Buffer.